packages/jetpack-ai-calypso/src/logo-generator/components/generator-modal.tsx

```diff
@@ -29,32 +31,69 @@ export const GeneratorModal: React.FC< GeneratorModalProps > = ( {
 	onClose,
 	siteDetails,
 } ) => {
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	const { setSiteDetails, fetchAiAssistantFeature } = useDispatch( STORE_NAME );
-	const [ isLoading, setIsLoading ] = useState( true );
-	const { selectedLogo } = useLogoGenerator();
+	const [ loadingState, setLoadingState ] = useState<
+		'loadingFeature' | 'analyzing' | 'generating' | null
+	>( null );
+	const [ initialPrompt, setInitialPrompt ] = useState< string | undefined >();
+	const [ isFirstCallOnOpen, setIsFirstCallOnOpen ] = useState( true );
+	const { selectedLogo, getAiAssistantFeature, generateFirstPrompt, generateLogo, logos } =
+		useLogoGenerator();
 	const siteId = siteDetails?.ID;
 
-	useEffect( () => {
-		if ( siteId ) {
-			setSiteDetails( siteDetails );
+	const getFeature = useCallback( async () => {
+		setLoadingState( 'loadingFeature' );
+		debug( 'Fetching AI assistant feature for site', siteId );
+		await fetchAiAssistantFeature( String( siteId ) );
+
+		// Wait for store to update.
+		return new Promise< Partial< AiFeatureProps > >( ( resolve ) => {
+			setTimeout( () => {
+				resolve( getAiAssistantFeature( String( siteId ) ) );
+			}, 100 );
+		} );
+	}, [ fetchAiAssistantFeature, getAiAssistantFeature, siteId ] );
+
+	const generateFirstLogo = useCallback( async () => {
+		try {
+			// First generate the prompt based on the site's data.
+			setLoadingState( 'analyzing' );
+			const prompt = await generateFirstPrompt();
+			setInitialPrompt( prompt );
+
+			// Then generate the logo based on the prompt.
+			setLoadingState( 'generating' );
+			await generateLogo( { prompt } );
+			setLoadingState( null );
+		} catch ( error ) {
+			debug( 'Error generating first logo', error );
+			setLoadingState( null );
 		}
-	}, [ siteId, siteDetails, setSiteDetails ] );
+	}, [ generateFirstPrompt, generateLogo ] );
 
-	useEffect( () => {
-		if ( isOpen ) {
-			if ( siteId ) {
-				debug( 'fetching ai assistant feature for site', siteId );
-				fetchAiAssistantFeature( String( siteId ) );
+	const handleModalOpen = useCallback( async () => {
+		// First fetch the feature data so we have the most up-to-date info from the backend.
+		try {
+			const feature = await getFeature();
+
+			// If there is any logo we do not need to generate a first logo again.
+			if ( logos.length > 0 ) {
```

Comment:
We'd need to remove the default logo I added on #86194 so to allow an empty history. LogoPresenter will suffer a bit